### PR TITLE
Keep dims in BatchNormTraining

### PR DESCRIPTION
**Pull request by denise-k**
This PR sets the `keepdims` option for the `mean` and `variance` calculations in `BatchNormTraining`. This prevents any dimension/broadcast mismatches in the batch norm calculation.

### CODE REVIEW

**Review by jd-bartlett96 — APPROVED** [review]
LGTM
